help_to_heat/frontdoor/interface.py

```diff
@@ -376,8 +376,9 @@ def parse_dpa_to_address(self, dpa):
         if "DEPENDENT_LOCALITY" in dpa:
             line_1_parts.append(dpa["DEPENDENT_LOCALITY"].title())
 
-        if "DEPENDENT_THOROUGHFARE_NAME" and "THOROUGHFARE_NAME" in dpa:
-            line_1_parts.append(dpa["THOROUGHFARE_NAME"].title())
+        if "DEPENDENT_THOROUGHFARE_NAME" not in dpa:
+            if "THOROUGHFARE_NAME" in dpa:
+                line_1_parts.append(dpa["THOROUGHFARE_NAME"].title())
         else:
             line_1_parts.append(dpa["DEPENDENT_THOROUGHFARE_NAME"].title())
             line_2_parts.insert(0, dpa["THOROUGHFARE_NAME"].title())
```

Comment:
Theoretically fails if we have dependent but not thoroughfare?

Reply:
indeed so. Sorry I feel like I've been crap fixing this bug